### PR TITLE
refactor importer part 2

### PR DESCRIPTION
**Pull request by obfusk**
Refactors the other importers to use a separate `saveAndDeduplicate()` function like #1422 did for the catima importer.

### CODE REVIEW

File: app/src/main/java/protect/card_locker/importexport/StocardImporter.java [hunk 1]
```diff
@@ -233,22 +274,39 @@ public void importData(Context context, SQLiteDatabase database, File inputFile,
                 headerColor = Utils.getHeaderColorFromImage(cardIcon, headerColor);
             }
 
-            long loyaltyCardInternalId = DBHelper.insertLoyaltyCard(database, store, note, null, null, BigDecimal.valueOf(0), null, cardId, null, barcodeType, headerColor, 0, null,0);
+            LoyaltyCard card = new LoyaltyCard(tempID, store, note, null, null, BigDecimal.valueOf(0), null, cardId, null, barcodeType, headerColor, 0, Utils.getUnixTime(), DBHelper.DEFAULT_ZOOM_LEVEL, 0);
```
Comment (TheLastProject) [inline app/src/main/java/protect/card_locker/importexport/StocardImporter.java:277]: Any reason why you changed the lastUsed from `0` to the current timestamp? The default DB value is `0`.

File: app/src/main/java/protect/card_locker/importexport/FidmeImporter.java [hunk 2]
```diff
@@ -132,6 +148,17 @@ private void importLoyaltyCard(Context context, SQLiteDatabase database, CSVReco
 
         // TODO: Front and back image
 
-        DBHelper.insertLoyaltyCard(database, store, note, null, null, BigDecimal.valueOf(0), null, cardId, null, barcodeType, headerColor, starStatus, null,archiveStatus);
+        // use -1 for the ID, it will be ignored when inserting the card into the DB
+        return new LoyaltyCard(-1, store, note, null, null, BigDecimal.valueOf(0), null, cardId, null, barcodeType, headerColor, starStatus, Utils.getUnixTime(), DBHelper.DEFAULT_ZOOM_LEVEL, archiveStatus);
```
Comment (TheLastProject) [inline app/src/main/java/protect/card_locker/importexport/FidmeImporter.java:152]: Any reason why you changed the lastUsed from `0` to the current timestamp? The default DB value is `0`.

**Comment by TheLastProject** [comment]
Looks good to me. I wasn't aware Voucher Vault's importer set "right now" as the "lastUsed" but Stocard and Fidme didn't. So that's a good consistency improvement.

**Comment by obfusk** [comment]
> I wasn't aware Voucher Vault's importer set "right now" as the "lastUsed" but Stocard and Fidme didn't.

They all did. Nothing really changed. But `DBHelper.insertLoyaltyCard()` takes a `Long` and uses `Utils.getUnixTime()` if it's `null`, whereas the `LoyaltyCard` constructor takes a `long`, which can't be `null`. So the inconsistency was in how it was calling `DBHelper.insertLoyaltyCard()`, but the result was the same before. I just couldn't pass `null` any more when using intermediate `LoyaltyCard` objects. The *code* is of course more consistent now :)

Of course the inconsistency between the arguments -- `Long` vs `long` -- remains. But I didn't want to change that in this PR. Let me know if you want me to follow up on that or just leave it as it is.

**Pull request merged**